### PR TITLE
feature/17000 Bug fixes

### PR DESCRIPTION
**Pull request by GeorgeChatzigiannis**
Validate pin for authentication and compare pinExpiresAt value with current date time

### CODE REVIEW

File: admin/services/generate-pins.service.js [hunk 1]
```diff
@@ -14,7 +18,7 @@ const generatePinService = {
     let pupils = await pupilDataService.getSortedPupils(schoolId, sortField, sortDirection)
     // filter pupils
     pupils = pupils
-      .filter(p => !p.pin && !p.attendanceCode && !p.result)
+      .filter(p => !generatePinService.isValidPin(p.pin, p.pinExpiresAt) && !p.attendanceCode && !p.result)
```
Comment (jon-shipley) [inline admin/services/generate-pins.service.js]: This whole line is a bit much. I would suggest moving this to a function - removePupilsWithPin or something

File: admin/services/generate-pins.service.js [hunk 2]
```diff
@@ -37,28 +41,28 @@ const generatePinService = {
...
     }
     return school
   },
 
-  isValidPin: (pinExpiresAt) => {
-    if (!pinExpiresAt) return false
-    return moment(pinExpiresAt).isBefore(moment().startOf('day').add(16, 'hours'))
+  isValidPin: (pin, pinExpiresAt) => {
+    if (!pinExpiresAt || !pin) return false
+    return moment(pinExpiresAt).isAfter(moment.utc())
```
Comment (GuyHarwood) [inline admin/services/generate-pins.service.js]: I would just pass the pupil into the method

Comment (GeorgeChatzigiannis) [reply admin/services/generate-pins.service.js]: @GuyHarwood all I need really is these two properties though.

File: admin/services/generate-pins.service.js (hunk 1 above)
Comment (GeorgeChatzigiannis) [reply admin/services/generate-pins.service.js]: @jon-shipley updated

**Pull request merged**